components/support/nimbus-fml/src/util/loaders.rs

```diff
@@ -31,7 +33,7 @@ impl LoaderConfig {
         if f.starts_with('@') {
             let parts = f.splitn(3, '/').collect::<Vec<&str>>();
             match parts.as_slice() {
-                [user, repo, path] => Some((format!("{user}/{repo}"), path.to_string())),
+                [user, repo, path] => Some((format!("{user}/{repo}"), (*path).to_owned())),
```

Comment:
What does this mean? This feels cleverer, but less clear than the `.to_string` it replaces.

Reply:
[.to_owned()](https://doc.rust-lang.org/stable/std/borrow/trait.ToOwned.html#tymethod.to_owned) is part of std::borrow::ToOwned, which basically just calls `String::from()` here. We have to use `(*path).to_owned()` because `path` is `&&str` here and only `&str` implements this.

I just learned while looking up sources that `str::to_string` is specialized as of *2016* so I've been operating on old information. For most types, calling `.to_string()` invokes `Display::fmt` (which used to be the case for `str`) which would involve a bunch of fmt machinery including potentially multiple allocations. Now calling `.into()`, `.to_owned()`, and `.to_string()` are all equivalent to `String::from`.

I'm used to calling .to_owned() on str since I've been writing Rust for so long, but now that I know its equivalent, I'm going to undo these .to_string() changes.